### ISSUE TITLE
Add `ItemCollection.to_collection`

### ISSUE THREAD

**Issue by gadomski** [stac-utils/pystac#1254]
This would be a nice convenience method that would generate the extents, etc from the items in the `ItemCollection`.

**Comment by jsignell** [comment]
I think of `ItemCollection`s as more analogous to `Catalog`s than to `Collection`s. I guess I think that because you can represent items from more than one `Collection` in an `ItemCollection`. But I guess from pystac's perspective Collections just inherit from Catalogs so maybe it doesn't matter?

**Comment by gadomski** [comment]
Yeah, I agree with the general point that not all `ItemCollection`s map to a `Collection`. However, often they do (e.g. a search result), and creating a "sub-collection" (here's a collection that describes my search results) could be of use to some folks?

**Comment by KeynesYouDigIt** [comment]
On paper, this makes sense as interesting and I was thinking of taking it on. However, do we have more bites from users on whether or not this would solve an active problem for them?

**Comment by gadomski** [comment]
Nope! This would be more of a "if you build it they will come" feature.

I do tend to create issues like this one when I'm using **pystac** and it's ergonomics don't match my expectations, so that's one user I guess 🙂

**Comment by jsignell** [comment]
I'm taking a crack at this. Considering doing it the other way around and putting a `from_items` on `Collection` that way it could take an iterable of items _or_ an `ItemCollection`. Is there a reason why the other way around would be better you think?
